ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -354,6 +357,27 @@ def _copy_and_remove_nans_from_obs(
             obs_with_no_nans.append(no_nan_obs)
         return obs_with_no_nans
 
+    def agent_count(
```

Comment:
I think this is a pretty complicated function but there are loops in the agent processor that have this information e.g. https://github.com/Unity-Technologies/ml-agents/blob/main/ml-agents/mlagents/trainers/agent_processor.py#L246. I think we can just add it to the buffer right there as an integer and then normalize in the trainer/network body

Reply:
cc @ervteng

Reply:
I'd be OK with any approach that doesn't call `get_masks_from_nans` again, as it's not the cheapest of functions for large inputs. I think we could move the call for this method into the `forward` method of MultiAgentNetworkBody perhaps, there we can re-use the mask that is already computed

Reply:
Adding it to the buffer seems redundant IMO, but it's not the worst

Reply:
I simplified the logic a bunch. I reuse `get_masks_form_nans` now.